### PR TITLE
go: update to 1.22.0

### PR DESCRIPTION
**Pull request by jiegec**
Topic Description
-----------------

- libguestfs: update to 1.52.0
- glide: bump REL due to go update
- dep: bump REL due to go update
- delve: update to 1.22.1
- rclone: update to 1.65.2
- hugo: update to 0.123.6
- hub: bump REL due to go update
- restic: update to 0.16.4
- pcstat: bump REL due to go update
- fzf: update to 0.46.1

... and 34 more commits

Package(s) Affected
-------------------

- rclone: 1.65.2
- hugo: 0.123.6
- git-lfs: 3.5.0
- gh: 2.44.1
- cf-tool: 1.0.0-4
- go-md2man: 2.0.3
- hub: 2.14.2+git20230617-1
- conmon: 2.1.10
- runc: 1.1.12
- snapd: 2.61.2
- direnv: 2.34.0
- restic: 0.16.4
- pcstat: 1:0.0.2-1
- fzf: 0.46.1
- automatic-component-toolkit: 1.6.0-1
- android-platform-tools: 34.0.4-1
- go: 1.22.0+tools0.18.0+net0.21.0
- dep: 0.5.4-17
- glide: 0.13.3-17
- delve: 1.22.1
- micro: 2.0.13-1
- liteide: 38.3-1
- obfs4proxy: 0.0.14
- geoipupdate: 6.1.0
- clash: 1.18.0-1
- dnscrypt: 2.1.5-1
- popub: 1:0+git20210814-2
- v2raya: 2.2.4
- kcptun: 20240107
- syncthing: 1.27.3
- graftcp: 1:0.7.2
- v2ray: 5.13.0
- frp: 0.54.0
- chisel: 1.9.1
- tailscale: 1.60.1
- kubo: 0.26.0
- v2ray-plugin: 1.3.2-1
- xray: 1.8.4
- gost: 2.11.5
- docker-compose: 2.24.6
- containerd: 1.7.13
- podman: 4.9.3+vsock0.7.2
- docker: 25.0.3+tini0.19.0
- libguestfs: 1.52.0

Security Update?
----------------

No

Build Order
-----------

```
#buildit go conmon go-md2man runc snapd containerd docker docker-compose podman cf-tool gh git-lfs android-platform-tools liteide micro chisel clash dnscrypt frp geoipupdate gost graftcp kcptun kubo obfs4proxy popub syncthing tailscale v2ray v2ray-plugin v2raya xray automatic-component-toolkit direnv fzf pcstat restic hub hugo rclone delve dep glide libguestfs
```

Test Build(s) Done
------------------

**Primary Architectures**

- [x] AMD64 `amd64`
- [x] AArch64 `arm64`
- [ ] LoongArch 64-bit `loongarch64`

**Secondary Architectures**

- [ ] Loongson 3 `loongson3`
- [x] PowerPC 64-bit (Little Endian) `ppc64el`
- [x] RISC-V 64-bit `riscv64`

**Experimental Architectures**

- [ ] MIPS R6 64-bit (Little Endian) `mips64r6el`

### CODE REVIEW

File: app-admin/conmon/spec [hunk 1]
```diff
@@ -1,4 +1,4 @@
-VER=2.1.8
+VER=2.1.10
 SRCS="https://github.com/containers/conmon/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-admin/conmon/spec]: Use Git tag here.

Comment (MingcongBai) [inline app-admin/conmon/spec]: Use Git tag here.

File: app-admin/runc/autobuild/defines [hunk 2]
```diff
@@ -3,3 +3,6 @@ PKGSEC=admin
 PKGDEP="criu libseccomp"
 BUILDDEP="go go-md2man"
 PKGDES="CLI tool for spawning and running containers according to the OCI specification"
+
+# Missing criu support
```
Comment (MingcongBai) [inline app-admin/runc/autobuild/defines]: Make this `# FIXME: Missing criu support"`.

File: app-containers/containerd/autobuild/defines [hunk 3]
```diff
@@ -5,3 +5,7 @@ BUILDDEP="go go-md2man"
 PKGDES="An industry-standard container runtime"
 
 ABSPLITDBG=0
+
+# Missing criu support
+FAIL_ARCH="riscv64"
+
```
Comment (MingcongBai) [inline app-containers/containerd/autobuild/defines]: Trailing new line.

File: app-containers/containerd/autobuild/defines [hunk 4]
```diff
@@ -5,3 +5,7 @@ BUILDDEP="go go-md2man"
 PKGDES="An industry-standard container runtime"
 
 ABSPLITDBG=0
+
+# Missing criu support
```
Comment (MingcongBai) [inline app-containers/containerd/autobuild/defines]: Make this `# FIXME: Missing criu support"`.

File: app-containers/docker/autobuild/defines [hunk 5]
```diff
@@ -6,4 +6,5 @@ PKGDES="A tool for deploying applications in containers"
 
 PKGPROV="moby"
 
-FAIL_ARCH="!(amd64|arm64|ppc64el|riscv64|loongarch64)"
+# Missing criu support
```
Comment (MingcongBai) [inline app-containers/docker/autobuild/defines]: Make this `# FIXME: Missing criu support"`.

File: app-containers/docker/autobuild/defines [hunk 6]
```diff
@@ -6,4 +6,5 @@ PKGDES="A tool for deploying applications in containers"
...
+FAIL_ARCH="!(amd64|arm64|ppc64el|loongarch64)"
```
Comment (MingcongBai) [inline app-containers/docker/autobuild/defines:10]: Hmm, this seems inconsistent with other instances of FAIL_ARCH for missing criu?

File: app-network/gost/autobuild/build [hunk 7]
```diff
@@ -1,10 +1,11 @@
+cd ..
 mkdir -p .gopath
-ln -sf "$SRCDIR"/gost-$PKGVER/vendor .gopath/src
-mkdir -p "$SRCDIR"/gost-$PKGVER/vendor/github.com/ginuerzh/
-ln -s "$SRCDIR"/gost-$PKGVER .gopath/src/github.com/ginuerzh/gost
-export GOPATH="$SRCDIR"/.gopath
+ln -sf "$SRCDIR"/vendor .gopath/src
```
Comment (MingcongBai) [inline app-network/gost/autobuild/build]: Please lint this script.

File: app-network/kcptun/autobuild/defines [hunk 8]
```diff
@@ -4,3 +4,5 @@ PKGDEP="glibc"
 BUILDDEP="go"
 PKGDES="An extremely simple UDP tunnel based on KCP"
 ABSPLITDBG=0
+
+FAIL_ARCH="loongarch64"
```
Comment (MingcongBai) [inline app-network/kcptun/autobuild/defines:9]: Why? Please leave a FIXME comment here.

File: app-network/kcptun/spec [hunk 9]
```diff
@@ -1,4 +1,4 @@
-VER=20230811
+VER=20240107
 SRCS="https://github.com/xtaci/kcptun/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-network/kcptun/spec]: Please use Git tag.

File: app-network/graftcp/spec [hunk 10]
```diff
@@ -1,5 +1,4 @@
-VER=0.4.0
+VER=0.7.2
 SRCS="https://github.com/hmgle/graftcp/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-network/graftcp/spec]: Please use Git tag.

File: app-network/gost/spec [hunk 11]
```diff
@@ -1,5 +1,4 @@
-VER=2.11.4
+VER=2.11.5
 SRCS="https://github.com/ginuerzh/gost/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-network/gost/spec]: Please use Git tag.

File: app-network/frp/spec [hunk 12]
```diff
@@ -1,4 +1,4 @@
-VER=0.51.3
+VER=0.54.0
 SRCS="tbl::https://github.com/fatedier/frp/archive/v${VER}.tar.gz"
```
Comment (MingcongBai) [inline app-network/frp/spec]: Please use Git tag.

File: app-network/geoipupdate/spec [hunk 13]
```diff
@@ -1,4 +1,4 @@
-VER=6.0.0
+VER=6.1.0
 SRCS="https://github.com/maxmind/geoipupdate/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-network/geoipupdate/spec]: Please use Git tag.

File: app-network/dnscrypt/spec [hunk 14]
```diff
@@ -1,4 +1,5 @@
 VER=2.1.5
+REL=1
 SRCS="https://github.com/jedisct1/dnscrypt-proxy/archive/$VER.tar.gz"
```
Comment (MingcongBai) [inline app-network/dnscrypt/spec]: Please use Git tag.

File: app-network/chisel/spec [hunk 15]
```diff
@@ -1,4 +1,4 @@
-VER="1.7.7"
+VER=1.9.1
 SRCS="https://github.com/jpillora/chisel/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-network/chisel/spec]: Please use Git tag.

File: app-devel/cf-tool/spec [hunk 16]
```diff
@@ -1,5 +1,5 @@
 VER=1.0.0
-REL=3
+REL=4
 SRCS="tbl::https://github.com/xalanq/cf-tool/archive/refs/tags/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-devel/cf-tool/spec]: Please use Git tag.

File: app-containers/docker-compose/spec [hunk 17]
```diff
@@ -1,4 +1,4 @@
-VER=2.23.3
+VER=2.24.6
 SRCS="tbl::https://github.com/docker/compose/archive/refs/tags/v${VER}.tar.gz"
```
Comment (MingcongBai) [inline app-containers/docker-compose/spec]: Please use Git tag.

File: app-network/kubo/spec [hunk 18]
```diff
@@ -1,5 +1,5 @@
-VER=0.18.1
+VER=0.26.0
 SRCS="tbl::https://github.com/ipfs/kubo/archive/refs/tags/v$VER.zip"
```
Comment (MingcongBai) [inline app-network/kubo/spec]: Please use Git tag.

File: app-network/popub/autobuild/build [hunk 19]
```diff
@@ -0,0 +1,6 @@
+abinfo "Building popub ..."
+make
+
+abinfo "Installing popub ..."
+make DESTDIR="$PKGDIR" PREFIX=/usr install
+
```
Comment (MingcongBai) [inline app-network/popub/autobuild/build]: Trailing new line.

File: app-network/v2ray-plugin/autobuild/defines [hunk 20]
```diff
@@ -4,3 +4,5 @@ BUILDDEP="go"
 PKGDEP="glibc"
 PKGSEC="net"
 ABSPLITDBG=0
+
+FAIL_ARCH="loongarch64"
```
Comment (MingcongBai) [inline app-network/v2ray-plugin/autobuild/defines:9]: Why? Please leave a FIXME comment here.

File: app-network/v2ray-plugin/spec [hunk 21]
```diff
@@ -1,4 +1,5 @@
 VER=1.3.2
+REL=1
 SRCS="https://github.com/shadowsocks/v2ray-plugin/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-network/v2ray-plugin/spec]: Please use Git tag.

File: app-network/v2raya/spec [hunk 22]
```diff
@@ -1,4 +1,4 @@
-VER=2.0.1
+VER=2.2.4
 SRCS="https://github.com/v2rayA/v2rayA/archive/refs/tags/v${VER}.tar.gz"
```
Comment (MingcongBai) [inline app-network/v2raya/spec]: Please use Git tag.

File: app-utils/automatic-component-toolkit/spec [hunk 23]
```diff
@@ -1,4 +1,5 @@
 VER="1.6.0"
+REL=1
 SRCS="https://github.com/Autodesk/AutomaticComponentToolkit/archive/refs/tags/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-utils/automatic-component-toolkit/spec]: Please use Git tag.

File: app-utils/direnv/spec [hunk 24]
```diff
@@ -1,4 +1,4 @@
-VER=2.32.3
+VER=2.34.0
 SRCS="https://github.com/direnv/direnv/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-utils/direnv/spec]: Please use Git tag.

File: app-utils/fzf/spec [hunk 25]
```diff
@@ -1,4 +1,4 @@
-VER=0.42.0
+VER=0.46.1
 SRCS="https://github.com/junegunn/fzf/archive/$VER.tar.gz"
```
Comment (MingcongBai) [inline app-utils/fzf/spec]: Please use Git tag.

File: app-utils/restic/spec [hunk 26]
```diff
@@ -1,4 +1,4 @@
-VER=0.16.0
+VER=0.16.4
 SRCS="https://github.com/restic/restic/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-utils/restic/spec]: Please use Git tag.

File: app-admin/conmon/spec (hunk 1 above)
Comment (jiegec) [reply app-admin/conmon/spec]: Fixed.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-admin/snapd/autobuild/defines [hunk 27]
```diff
@@ -6,3 +6,4 @@ PKGDEP="squashfs-tools libseccomp systemd apparmor xdg-desktop-portal \
 BUILDDEP="go libcap xfsprogs docutils"
 
 ABQA=0
+FAIL_ARCH="!(amd64|arm64|ppc64el|riscv64)"
```
Comment (MingcongBai) [inline app-admin/snapd/autobuild/defines:11]: Why? Please leave a FIXME comment here.

File: app-web/hugo/spec [hunk 28]
```diff
@@ -1,4 +1,4 @@
-VER=0.118.2
+VER=0.123.6
 SRCS="https://github.com/gohugoio/hugo/archive/v$VER.tar.gz"
```
Comment (MingcongBai) [inline app-web/hugo/spec]: Please use Git tag.

**Review by MingcongBai — CHANGES_REQUESTED** [review]

File: app-network/gost/autobuild/build (hunk 7 above)
Comment (MingcongBai) [reply app-network/gost/autobuild/build]: Please use verbosity.

File: app-network/obfs4proxy/autobuild/build [hunk 29]
```diff
@@ -1,16 +1,5 @@
...
-export GO111MODULE=auto
-
 abinfo "Building obfs4proxy ..."
-(
-    cd "$SRCDIR"/obfs4proxy
-    GOPATH="$SRCDIR" \
-    GOBIN="$PWD" \
-        go get -v
-)
+go build -o obfs4proxy/obfs4proxy ./obfs4proxy
```
Comment (MingcongBai) [inline app-network/obfs4proxy/autobuild/build]: Absolute path?

**Review by MingcongBai — CHANGES_REQUESTED** [review]

**Review by MingcongBai — APPROVED** [review]

**Review by MingcongBai — APPROVED** [review]

**Pull request merged**